pkg/authz/entry_point_authorizer.go

```diff
@@ -0,0 +1,335 @@
+package authz
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/bacalhau-project/bacalhau/pkg/config/types"
+	"github.com/bacalhau-project/bacalhau/pkg/publicapi/apimodels"
+	"github.com/pkg/errors"
+)
+
+const (
+	MinimumPasswordLength = 10
+	MaximumPasswordLength = 255
+	MinimumAPIKeyLength   = 20
+	MaximumAPIKeyLength   = 255
+	UsernameMaxLength     = 100
+)
+
+type entryPointAuthorizer struct {
+	nodeID string
+	// Separate maps for different authentication methods
+	basicAuthUsers map[string]types.AuthUser // Key is username
+	apiKeyUsers    map[string]types.AuthUser // Key is API key
+	// Capability checker for authorization
+	capabilityChecker *CapabilityChecker
+	// Endpoint permissions mapping
+	endpointPermissions map[string]string
+
+	// Specialized Authorizers
+	basicAuthAuthorizer Authorizer
+	apiKeyAuthorizer    Authorizer
+	jwtAuthorizer       Authorizer
+}
+
+// validateUser validates a user configuration and returns an error if it's invalid
+func (a *entryPointAuthorizer) validateUser(user types.AuthUser) error {
+	// Validation: Check that user has a non-empty Alias
+	if user.Alias == "" {
+		return fmt.Errorf("user has empty Alias, Alias is mandatory")
+	}
```

Comment:
I thought alias is optional?

Reply:
It is optional now.
Made it mandatory for quick testing while working on the PR.